From d95a95ffcdca931e8752de4af8c0009011e8fb59 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 08 Aug 2021 20:31:30 +0300
Subject: [PATCH] [SW-52069] gaudi: security: refine sync mngr prot

Change the Sync Manager protection to the whole page - do not leave any
registers unprotected.

Change-Id: I3e5c9ef873df35daf9ce2408a09910aed33cbf28
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index eab1142..fb57727 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -609,23 +609,12 @@
 #define SM_ENGINES	4
 static void preboot_secure_sm(void)
 {
-	u32 prot_base, base, mask;
+	u32 base;
 	int i;
-
-	mask = ~(PROT_BIT(mmSYNC_MNGR_W_S_SYNC_MNGR_GLBL_SM_SEI_MASK) &
-		 PROT_BIT(mmSYNC_MNGR_W_S_SYNC_MNGR_GLBL_SM_SEI_CAUSE));
 
 	for (i = 0; i < SM_ENGINES; i++) {
 		base = (mmSYNC_MNGR_GLBL_W_S_BASE & ~HL_CFG_BASE) + i * SM_DIST;
-
-		/* offsets 0x0 - 0x80 */
-		prot_base = base;
-		regmap_writel(mask, PRIV_PROT_REG(prot_base));
-
-		/* offsets 0x80 - 0xF80 */
-		prot_base += SNGL_REG_PROT_SIZE;
-		set_prot_bits_range_all(prot_base,
-					PROT_SIZE_MAX - (prot_base - base));
+		set_prot_bits_range_all(base, PROT_SIZE_MAX);
 	}
 }
 
